blockchain/genesis.go

```diff
@@ -284,7 +284,14 @@ func (g *Genesis) ToBlock(baseStateRoot common.Hash, db database.DBManager) *typ
 	stateDB, _ := state.New(baseStateRoot, state.NewDatabase(db))
 	for addr, account := range g.Alloc {
 		if len(account.Code) != 0 {
+			originalCode := stateDB.GetCode(addr)
 			stateDB.SetCode(addr, account.Code)
+			// If originalCode is not nil and baseStateRoot is not an empty hash,
+			// just update the code and don't change the other states
+			if originalCode != nil && !common.EmptyHash(baseStateRoot) {
```

Comment:
I think `!common.EmptyHash(baseStateRoot)` condition is not necessary. Isn't `originalCode` always `nil` if `baseStateRoot` is empty hash?

Reply:
You're right, it happens only if the address appears more than once in genesis.json.
Let's remove that condition.